src/Adapter/Presenter/Product/ProductLazyArray.php

```diff
@@ -382,22 +382,25 @@ public function getGroupedFeatures()
     }
 
     /**
+     * See https://support.google.com/merchants/answer/6324448?hl=en for more information
+     *
      * @arrayAccess
      *
      * @return string
      */
     public function getSeoAvailability()
     {
-        $seoAvailability = 'https://schema.org/';
-        if ($this->product['quantity'] > 0) {
-            $seoAvailability .= 'InStock';
-        } elseif ($this->product['quantity'] <= 0 && $this->product['allow_oosp']) {
-            $seoAvailability .= 'PreOrder';
+        // This case doesn't show by default, if one wants to keep inactive products
+        // displayed as discontinued, he will have proper structured data
+        if ($this->product['active'] === 0) {
+            return 'https://schema.org/Discontinued';
```

Comment:
Do we really need it? The inactive product should not be visible in the front office, right? Am I missing something?

Reply:
Yes, they are not visible now, but I want to add this possibility in the future. More details here - https://github.com/PrestaShop/PrestaShop/issues/23533

Reply:
Yes, it will, I want to implement an option to send 200/404/410 along with "DISCONTINUED" text.

Reply:
@kpodemski WIP - https://github.com/PrestaShop/PrestaShop/pull/29720

Reply:
@kpodemski Removed from THIS PR